### PR TITLE
Feature/0015 kafka lib

### PR DESCRIPTION
**Pull request by ajax-torbin-v**

### CODE REVIEW

File: domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusConsumerForNats.kt [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
+) : KafkaHandler<UpdateOrderStatusResponse, TopicSingle> {
+    override val groupId: String = UPDATE_STATUS_CONSUMER_GROUP_NATS
+    override val parser: Parser<UpdateOrderStatusResponse> = UpdateOrderStatusResponse.parser()
+    override val subscriptionTopics: TopicSingle = TopicSingle(KafkaTopic.KafkaOrderStatusUpdateEvents.UPDATE)
+
+    override fun handle(kafkaEvent: KafkaEvent<UpdateOrderStatusResponse>): Mono<Unit> {
+        return publisher.publish(
+            NatsSubject.Order.getUpdateStatusByUserId(kafkaEvent.data.success.order.userId),
+            kafkaEvent.data
+        )
```
Comment (ajax-butriy-v) [inline domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusConsumerForNats.kt]: do ack

File: domainservice/src/test/kotlin/com/example/delivery/kafka/OrderUpdateStatusProcessorTest.kt [hunk 2]
```diff
@@ -29,46 +30,35 @@ class KafkaTest : AbstractIntegrationTest() {
...
-            .atMost(15, TimeUnit.SECONDS)
-            .untilAsserted {
-                assertTrue(receivedMessages.contains(expected))
-            }
+        val eventProvider = kafkaMockExtension.listen<OrderStatusUpdateNotification>(
+            KafkaTopic.KafkaOrderStatusUpdateEvents.NOTIFICATIONS,
+            OrderStatusUpdateNotification.parser()
+        )
+
+        val event = eventProvider.awaitFirst({ it.toBuilder().clearTimestamp().build() == expected })
```
Comment (ajax-butriy-v) [inline domainservice/src/test/kotlin/com/example/delivery/kafka/OrderUpdateStatusProcessorTest.kt:55]: Clock?

File: domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusNotificationProducer.kt [hunk 3]
```diff
@@ -2,30 +2,20 @@ package com.example.delivery.kafka
...
+    private val kafkaPublisher: KafkaPublisher,
 ) {
     fun notify(notification: OrderStatusUpdateNotification): Mono<Unit> {
-        return updateStatusNotificationSender.send(
-            createMessage(notification)
+        return kafkaPublisher.publish(
+            KafkaTopic.KafkaOrderStatusUpdateEvents.NOTIFICATIONS,
+            notification.userId,
+            notification.toByteArray()
         ).then(Unit.toMono())
```
Comment (ajax-butriy-v) [inline domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusNotificationProducer.kt]: thenReturn
```suggestion
            notification.toByteArray()
        ).thenReturn(Unit)
```

Comment (ajax-torbin-v) [reply domainservice/src/main/kotlin/com/example/delivery/kafka/OrderUpdateStatusNotificationProducer.kt:19]: Referred to [docs](https://youtu.be/ooOELrGMn14?si=TDg6r6eXb39oYKfU)

**Review by ajax-shulepa-a — APPROVED** [review]

**Pull request merged**